mlir/include/mlir/Tools/lsp-server-support/Protocol.h

```diff
@@ -677,6 +677,15 @@ enum class DiagnosticSeverity {
   Hint = 4
 };
 
+enum class DiagnosticTag {
+  Unnecessary = 1,
+  Deprecated = 2,
+};
+
```

Comment:
Just for consistency, add a `/// Add support for JSON serialization.` line before these json methods. The rest of the file is organized this way.